ckanext/ddi/tests/test_blueprints.py

```diff
@@ -0,0 +1,158 @@
+            status=200,
+        )
+        _assert_in_body('<input id="field-upload" type="file" name="upload"', resp)
+        _assert_in_body(
+            '<input id="field-rdf_upload" type="file" name="rdf_upload"', resp
+        )
+
+    def test_form_submit_success_xml_file_from_upload(
+        self, app, monkeypatch, tmpdir, ckan_config
+    ):
+        _patch_storage_path(monkeypatch, tmpdir, ckan_config)
+        files = {'upload': 'ddi_test.xml'}
+        resp = _post_request(
+            app, '/dataset/import', {}, files, self.extra_environ, status=302
+        )
+        expected_id = 'ddi-test-1'
+        try:
+            toolkit.requires_ckan_version("2.9")
+            assert (
+                '/dataset/{}/resource/new'.format(expected_id)
+                in resp.headers['location']
+            )
+        except toolkit.CkanVersionException:
+            assert (
+                '/dataset/new_resource/{}'.format(expected_id)
+                in resp.headers['location']
+            )
+        dataset = toolkit.get_action('package_show')(
+            {'ignore_auth': True}, {'id': expected_id}
+        )
+        assert dataset
+        assert len(dataset['resources']) == 1
+        assert 'ddi_test.xml' in dataset['resources'][0]['url']
+
+    def test_form_submit_success_xml_and_rdf_files_from_uploads(
+        self, app, monkeypatch, tmpdir, ckan_config
+    ):
+        _patch_storage_path(monkeypatch, tmpdir, ckan_config)
+        files = {'upload': 'ddi_test.xml', 'rdf_upload': 'ddi_test.rdf'}
+        resp = _post_request(
+            app, '/dataset/import', {}, files, self.extra_environ, status=302
+        )
+        expected_id = 'ddi-test-1'
+        try:
+            toolkit.requires_ckan_version("2.9")
+            assert (
+                '/dataset/{}/resource/new'.format(expected_id)
+                in resp.headers['location']
+            )
+        except toolkit.CkanVersionException:
+            assert (
+                '/dataset/new_resource/{}'.format(expected_id)
+                in resp.headers['location']
+            )
+        dataset = toolkit.get_action('package_show')(
+            {'ignore_auth': True}, {'id': expected_id}
+        )
+        assert dataset
+        assert len(dataset['resources']) == 2
+        assert 'ddi_test.xml' in dataset['resources'][0]['url']
+        assert 'ddi_test.rdf' in dataset['resources'][1]['url']
+
+    """
+    def test_form_submit_success_xml_file_from_url(
+        self, app, monkeypatch, tmpdir, ckan_config
+    ):
+        _patch_storage_path(monkeypatch, tmpdir, ckan_config)
+        # TODO: test importing from a URL, mock the HTTP requests with responses
+
+    def test_form_submit_success_xml_and_rdf_files_from_urls(
+        self, app, monkeypatch, tmpdir, ckan_config
+    ):
+        _patch_storage_path(monkeypatch, tmpdir, ckan_config)
+        # TODO: test importing from URLs, mock the HTTP requests with responses
+    """
```

Comment:
As noted in the top post, there's a bunch of stuff I haven't tested, but I didn't want this to turn into too much of a timesink. I have to stop pulling this thread somewhere. At least having got some tests written and running under CI we've got a starting point to improve from if we want to..

Reply:
Sounds good, I think what we have now is comprehensive enough.